request.ts

```diff
@@ -0,0 +1,126 @@
+import * as _ from 'lodash'
+import * as request from 'request'
+import * as Promise from 'bluebird'
+import { PinejsClientCoreFactory } from 'pinejs-client-core'
+import TypedError = require('typed-error')
+
+interface Cache {
+	get(url: string): Promise<CachedResponse>
+	set(url: string, value: CachedResponse): void
+}
+interface BluebirdLRU {
+	(params: { [index: string]: any }): Cache
+	NoSuchKeyError: TypedError
+}
+const BluebirdLRU: BluebirdLRU = require('bluebird-lru-cache')
+
+const requestAsync = Promise.promisify(request)
+
+class StatusError extends TypedError {
+	constructor(
+		message: string,
+		public statusCode: number
+	) {
+		super(message)
+	}
+}
+
+interface CachedResponse {
+	etag: string
+	body: any
+}
+
+export interface BackendParams {
+	cache?: {
+		[index: string]: any
+	}
+}
+type PromiseObj = Promise<{}>
+
+const validParams: Array<keyof BackendParams> = ['cache']
+const PinejsClientCore = PinejsClientCoreFactory(_ as any as PinejsClientCoreFactory.Util, Promise)
+export class PinejsClientRequest extends PinejsClientCore<PinejsClientRequest, PromiseObj, Promise<PinejsClientCoreFactory.PromiseResultTypes>> {
+	backendParams: BackendParams = {}
+	private cache?: Cache
+
+	constructor(params: PinejsClientCoreFactory.Params, backendParams?: BackendParams) {
+		super(params)
+		if (backendParams != null && _.isObject(backendParams)) {
+			for (const v in validParams) {
+				const validParam = v as keyof BackendParams
```

Comment:
I think that this `for in` should actually be a `for of`, [otherwise `v` would hold the indexes of the array](https://www.typescriptlang.org/play/index.html#src=const%20a%20%3D%20%5B'cache'%5D%3B%0D%0A%0D%0Afor%20(const%20x%20in%20a)%20%7B%0D%0A%20%20%20%20console.log('in'%2C%20x)%0D%0A%7D%0D%0A%0D%0Afor%20(const%20x%20of%20a)%20%7B%0D%0A%20%20%20%20console.log('of'%2C%20x)%0D%0A%7D%0D%0A).
That way we can also get rid of the `v as keyof BackendParams`.

Otherwise, this might get cleaner if we rewrite it as:
```ts
_.assign(this.backendParams, _.pick(backendParams, validParams))
```

Reply:
Yes, you're completely right - good catch.  And I do like the lodash version, I was used to not being able to assume lodash is available in the pinejs-client-core but we do have it here :D